File: sample/src/main/java/com/davemorrissey/labs/subscaleview/test/eventhandling/EventHandlingActivity.java
package com.davemorrissey.labs.subscaleview.test.eventhandling;

import static com.davemorrissey.labs.subscaleview.test.R.layout.pages_activity;
import static com.davemorrissey.labs.subscaleview.test.R.string.event_p1_subtitle;
import static com.davemorrissey.labs.subscaleview.test.R.string.event_p1_text;
import static com.davemorrissey.labs.subscaleview.test.R.string.event_p2_subtitle;
import static com.davemorrissey.labs.subscaleview.test.R.string.event_p2_text;
import static com.davemorrissey.labs.subscaleview.test.R.string.event_p3_subtitle;
import static com.davemorrissey.labs.subscaleview.test.R.string.event_p3_text;
import static com.davemorrissey.labs.subscaleview.test.R.string.event_title;

import android.os.Bundle;
import android.widget.Toast;

import com.davemorrissey.labs.subscaleview.ImageSource;
import com.davemorrissey.labs.subscaleview.SubsamplingScaleImageView;
import com.davemorrissey.labs.subscaleview.test.AbstractPagesActivity;
import com.davemorrissey.labs.subscaleview.test.Page;
import com.davemorrissey.labs.subscaleview.test.R.id;

import java.util.Arrays;

public class EventHandlingActivity extends AbstractPagesActivity {

    public EventHandlingActivity() {
        super(event_title, pages_activity, Arrays.asList(
                new Page(event_p1_subtitle, event_p1_text),
                new Page(event_p2_subtitle, event_p2_text),
                new Page(event_p3_subtitle, event_p3_text)
        ));
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        SubsamplingScaleImageView imageView = findViewById(id.imageView);
        imageView.setImage(ImageSource.asset(this, "sanmartino.jpg"));
        imageView.setOnClickListener(v -> Toast.makeText(v.getContext(), "Clicked", Toast.LENGTH_SHORT).show());
        imageView.setOnLongClickListener(v -> {
            Toast.makeText(v.getContext(), "Long clicked", Toast.LENGTH_SHORT).show();
            return true;
        });
    }
}

File: sample/src/main/kotlin/com/davemorrissey/labs/subscaleview/test/eventhandling/EventHandlingActivity.kt
package com.davemorrissey.labs.subscaleview.test.eventhandling

import android.os.Bundle
import android.view.View
import android.widget.Toast
import com.davemorrissey.labs.subscaleview.ImageSource
import com.davemorrissey.labs.subscaleview.SubsamplingScaleImageView
import com.davemorrissey.labs.subscaleview.test.AbstractPagesActivity
import com.davemorrissey.labs.subscaleview.test.Page
import com.davemorrissey.labs.subscaleview.test.R.id
import com.davemorrissey.labs.subscaleview.test.R.layout
import com.davemorrissey.labs.subscaleview.test.R.string

class EventHandlingActivity : AbstractPagesActivity(
    string.event_title, layout.pages_activity, listOf(
        Page(string.event_p1_subtitle, string.event_p1_text),
        Page(string.event_p2_subtitle, string.event_p2_text),
        Page(string.event_p3_subtitle, string.event_p3_text)
    )
) {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val imageView = findViewById<SubsamplingScaleImageView>(id.imageView)
        imageView.setImage(ImageSource.asset(this, "sanmartino.jpg"))
        imageView.setOnClickListener { v: View ->
            Toast.makeText(
                v.context,
                "Clicked",
                Toast.LENGTH_SHORT
            ).show()
        }
        imageView.setOnLongClickListener { v: View ->
            Toast.makeText(v.context, "Long clicked", Toast.LENGTH_SHORT).show()
            true
        }
    }
}